app/models/concerns/fiscal_year_mixin.rb

```diff
@@ -9,5 +11,11 @@ def self.which_fiscal_year(year, month)
         year
       end
     end
+
+    def self.for_fiscal_year(year)
```

Comment:
How about a refactor so that the class method part is in the mixin and the scope definition is still in the AR classes themselves? The date math can be part of the mixin.